keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resource/ClusterHandler.kt

```diff
@@ -161,6 +120,26 @@ class ClusterHandler(
       .getCorrelatedExecutions(name.value)
       .isNotEmpty()
 
+  override suspend fun onActuationComplete(resource: Resource<ClusterSpec>, current: Cluster) {
```

Comment:
This method on `ClusterHandler` lets me update the artifact version known to exist in an environment via `PromotionRepository`. I'm reverse engineering the version from the image id in the current model.